README.md

```diff
@@ -31,6 +39,131 @@ Create a `.env.docker` file to store your own custom env vars. See [`example.doc
+further clarifications on backward compatibility guarantees:
+
+**Go API compatibility**
+
+GoTrue is not meant to be used as a Go library. There are no guarantees on
+backward API compatibility when used this way regardless which version number
+changes.
+
+**Patch**
+
+Changes to the patch version guarantees backward compatibility with:
+
+- Database objects (tables, columns, indexes, functions).
+- REST API
+- JWT structure
+- Configuration
+
+Guaranteed examples:
+
+- A column won't change its type.
+- A table won't change its primary key.
+- An index will not be removed.
+- A uniqueness constraint will not be removed.
+- A REST API will not be removed.
+- Parameters to REST APIs will work equivalently as before (or better, if a bug
+  has been fixed).
+- Configuration will not change.
+
+Not guaranteed examples:
+
+- A table may add new columns.
+- Columns in a table may be reordered.
+- Non-unique constraints may be removed (database level checks, null, default
+  values).
+- JWT may add new properties.
+
+**Minor**
+
+Changes to minor version guarantees backward compatibility with:
+
+- REST API
+- JWT structure
+- Configuration
+
+Exceptions to these guarantees will be made only when serious security issues
+are found that can't be remedied in any other way.
+
+Guaranteed examples:
+
+- Existing APIs may be deprecated but continue working for the next few minor
+  version releases.
+- Configuration changes may become deprecated but continue working for the next
+  few minor version releases.
+- Already issued JWTs will be accepted, but new JWTs may be with a different
+  structure (but usually similar).
+
+Not guaranteed examples:
+
+- Removal of JWT fields after a deprecation notice.
+- Removal of certain APIs after a deprecation notice.
+- Removal of sign-in with external providers, after a deprecation notice.
+- Deletion, truncation, significant schema changes to tables, indexes, views,
+  functions.
+
+We aim to provide a deprecation notice in execution logs for at least two major
+version releases or two weeks if multiple releases go out. Compatibility will
+be guaranteed while the notice is live.
+
+**Major**
+
+Changes to the major version do not guarantee any backward compatibility with
+previous versions.
+
+### Inherited features
+
+Certain inherited features from the Netlify codebase are not supported by
+Supabase and they may be removed without prior notice in the future. This is a
+comprehensive list of those features:
+
+1. Multi-tenancy via the `instances` table i.e. `GOTRUE_MULTI_INSTANCE_MODE`
+   configuration parameter.
+2. System user (zero UUID user).
+3. Super admin via the `is_super_admin` column.
+4. SAML sign-in provider via the `GOTRUE_SAML_ENABLED` configuration
+   parameter. (A different implementation for SAML may appear in the future
+   which will be supported.)
+5. Support for MySQL based databases. (Only Postgres is supported.)
```

Comment:
Hmm are we using any Postgres specific features which would completely block a user from self-hosting on MySQL?

If not we could consider stating that it's possible but not natively supported (e.g. we won't have the starter files and we won't be taking queries about self-hosting with MySQL but the dedicated user could self-host with MySQL if they added additional code). 

This defers the decision of tying GoTrue to Postgres and allows us to eventually come back and support other Databases if we want to. At the same time, I realise it is quite unlikely that we will want to support other DBs given that the rest of Supabase tooling is heavily dependent on Postgres.


```suggestion
5. Built-in support for MySQL based databases. (We primarily support Postgres)
```

Reply:
imo @hf 's wording is sane. If we want to add support for other DBs in the future that's fine, we can always come back and update this.

Reply:
Ok! Resolving